pkg/reconciler/resource.go

```diff
@@ -122,20 +134,53 @@ type GenericResourceReconciler struct {
 
 type ResourceReconcilerOption func(*ReconcilerOpts)
 
+type RecreateResourceCondition func(kind schema.GroupVersionKind, status metav1.Status) bool
+
 // Recommended to use NewReconcilerWith + ResourceReconcilerOptions
 type ReconcilerOpts struct {
-	Log                                              logr.Logger
-	Scheme                                           *runtime.Scheme
-	EnableRecreateWorkloadOnImmutableFieldChange     bool
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+	// Enable recreating workloads and services when the API server rejects an update
+	EnableRecreateWorkloadOnImmutableFieldChange bool
+	// Custom log message to help when a workload or service need to be recreated
 	EnableRecreateWorkloadOnImmutableFieldChangeHelp string
+	// The delay in seconds to wait before checking back after deleted the resource (10s by default)
+	RecreateRequeueDelay *int32
+	// List of callbacks evaluated to decide whether a given gvk is enabled to be recreated or not
+	RecreateEnabledResourceCondition RecreateResourceCondition
+	// Immediately recreate the resource instead of deleting and returning with a requeue
+	RecreateImmediately bool
+	// Check the update error message contains this substring before recreate. Default: "immutable"
+	RecreateErrorMessageContains *string
```

Comment:
Maybe like this? I'm fine this naming as well, just an idea.
```suggestion
	RecreateErrorMessageSubstring *string
```